helix-core/src/syntax.rs

```diff
@@ -60,8 +60,11 @@ fn default_timeout() -> u64 {
 }
 
 #[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
 pub struct Configuration {
     pub language: Vec<LanguageConfiguration>,
+    #[serde(default)]
+    pub language_server: HashMap<String, LanguageServerConfiguration>,
 }
```

Comment:
Just wondering, do we need a global LSP injection? Given that people are asking for copilot https://github.com/helix-editor/helix/discussions/4037 and for @beyondCritics to be a serious coder, I think we might want to have a global LSP configurations to apply a single LSP for all languages? Otherwise people might have to add something like copilot to each language.

Reply:
I thought about that too for example for ltex-ls (it works for comments). But I think that should be a followup PR. Almost all LSPs out there are currently constrained to a fairly small set of languages. I wouldn't want to rush the design for that cinfig option

Reply:
Yeah I agree, that's a good idea, but it can be a follow up PR